### ISSUE TITLE
authenticate: error used to deny auth crashes server

### ISSUE THREAD

**Issue by goliatone** [primus/primus#715]
**Version:**
primus: ^7.3.2
primus-emit: ^1.0.0
fortress-maximus: 0.0.5
ws: ^6.1.2

**Environment:**
  - **Operating system:** macos
  - **Browser:** chrome
  - **Node.js:** v10.14.2

**Expected result:** 
* Denied connections should be handled by `primus.on('outgoing::open')`. 
* `primus.on('error')` should be notified? 

**Actual result:**
Server crashes due to **read ECONNRESET** error.

```
  ERROR [22:27:22:682] ├── console      : ioauth: Missing access token
 VERBOSE[22:27:22:683] ├── console      : [Object: null prototype] {
                                        _primuscb: 'Mb2F4-d' }
  WARN  [22:27:22:692] ├── console      : SERENE: Got "uncaughtException", start shudown process
  WARN  [22:27:22:692] ├── console      : SERENE: Timeout will execute in 1 second from now
  ERROR [22:27:22:693] ├── console      : SERENE: Exit due an uncaught exception, read ECONNRESET
  ERROR [22:27:22:693] ├── console      : Error: read ECONNRESET
                                         at TCP.onStreamRead (internal/stream_base_commons.js:111:27)
                                     ---------------------------------------------
                                         at _destroy (internal/streams/destroy.js:34:15)
                                         at Socket._destroy (net.js:617:3)
                                         at Socket.destroy (internal/streams/destroy.js:32:8)
                                         at TCP.onStreamRead (internal/stream_base_commons.js:111:19)
```

**Steps to reproduce:**
~~If you omit the `transformer: 'websockets'` in the **Primus** constructor.~~
Trying to figure out how to fix/reproduce

**Comment by lpinca** [comment]
What version of ws are you using?

**Comment by goliatone** [comment]
Updated the original comment with the version: **ws: ^6.1.2**

**Comment by ShittyAdvice** [comment]
I experience the same issue and it seems to only be the case with node v10.*.
Switching to node v8.16.0 fixes the issue for now.

**Comment by lpinca** [comment]
This is probably caused by this breaking change https://github.com/nodejs/node/pull/18868.

@ShittyAdvice's workaround is ok if there are no middleware before the auth middleware.

A more robust fix would be adding a listener for the `'error'` event before this line https://github.com/primus/primus/blob/ae78c1e3ed82ab84e98bb56e3bc80721048c29a6/transformer.js#L191

```js
socket.on('error', function() {
  this.destroy();
});
```

This restores Node.js 8 default behavior.

**Comment by lpinca** [comment]
I've released `primus@7.3.3`. Let me know if the issue persists.
